src/Strategy.js

```diff
@@ -0,0 +1,73 @@
+//@flow
+
+import {expandStyle} from './compiler';
+import {Spec} from './ElementPropSpec';
+import {staticStylesheetManager, dynamicStylesheetManager} from './StylesheetManager';
+
+export const resolve = ({
+  defaultClassName,
+  defaultComponent,
+  ownProps,
+}: {
+  defaultClassName: Array<string>,
+  defaultComponent: *,
+  ownProps: *,
+}) => {
+  let Component = defaultComponent;
+  const className = defaultClassName;
```

Comment:
So what we should do here? move to `defaultProps` instead `this.constructor` way?